app/src/main/java/com/bridgewell/quickstart/android/activity/TabbedViewActivity.kt

```diff
@@ -0,0 +1,191 @@
+                showToast("onAdLoaded")
+            }
+
+            override fun onAdViewDisplayed(bannerView: BwsAdView?) {
+                showToast("onAdDisplayed")
+            }
+
+            override fun onAdViewFailed(bannerView: BwsAdView?, exception: AdException?) {
+                showToast("onAdFailed ${exception?.message}")
+                Timber.d("onAdFailed ${exception?.message}")
+            }
+
+            override fun onAdViewClicked(bannerView: BwsAdView?) {
+                showToast("onAdClicked")
+            }
+
+            override fun onAdModalBrowserClosed(bannerView: BwsAdView?) {
+                showToast("onAdModalBrowserClosed")
+            }
+
+            override fun onAdViewClosed() {
+                showToast("onAdViewClosed")
+            }
+        }
+
+        when (adType) {
+            AdType.POP_UP_AD -> {
+                inAppApi.createBwsPopupAd(
+                    this,
+                    configID = CONFIG_ID_BANNER,
+                    refreshTimeSeconds = 0,
+                    listener = listener
+                )
+            }
+            AdType.STICKY_BOTTOM_RIGHT -> {
+                inAppApi.createBwsRightSideStickyAd(
+                    this,
+                    configID = CONFIG_ID_BANNER,
+                    bottomMargin = 77,
+                    refreshTimeSeconds = 0,
+                    listener = listener
+                )
+            }
+            AdType.MOBILE_STICKY_BOTTOM -> {
+                inAppApi.createBwsMobileStickyAd(
+                    this,
+                    configID = CONFIG_ID_MOBILE_STICKY_BOTTOM,
+                    bottomMargin = 65,
+                    refreshTimeSeconds = 0,
+                    listener = listener
+                )
+            }
+            AdType.BANNER -> {
+                val skeletonContainer = findViewById<LinearLayout>(R.id.topSkeletonContainer)
+                skeletonContainer.visibility = View.VISIBLE
+                for (i in 1..6) {
+                    val imageViewId = resources.getIdentifier("topImageView$i", "id", packageName)
+                    val imageView = findViewById<ImageView>(imageViewId)
+                    val skeletonColor = when (i) {
+                        1, 2 -> R.color.skeleton_gray_1
+                        else -> R.color.skeleton_gray_2
+                    }
+                    imageView.loadSkeleton {
+                        color(skeletonColor)
+                        cornerRadius(4F)
+                    }
+                }
+                val adWrapperView = findViewById<FrameLayout>(R.id.frameAdWrapper)
+                inAppApi.createBwsBannerAd(
+                    this,
+                    configID = CONFIG_ID_BANNER,
+                    width = adWrapperView.width,
+                    height = adWrapperView.height,
+                    viewContainer = adWrapperView,
+                    listener = listener
```

Comment:
![image](https://github.com/user-attachments/assets/4c16f621-ccbb-47db-b126-44338aacd806)

Reply:
This issue will be resolved after the project is synced